nexus/networking/src/firewall_rules.rs

```diff
@@ -345,6 +353,45 @@ pub async fn resolve_firewall_rules_for_sled_agent(
             priority: rule.priority.0 .0,
         });
     }
+
+    // Add rules that implement our IP allowlist for user-facing services.
+    //
+    // These rules are implicit, and not stored in the firewall rule table,
+    // since they're logically a different thing. We'll convert them into
+    // sled-agent firewall rules here and insert them.
+    //
+    // Note that this only applies to the internal services VPC, so we ignore
+    // these rules if we have no such interfaces for that.
+    if vpc.id() == *SERVICES_VPC_ID {
```

Comment:
This is a vague concern, feel free to defer to urgency here, or ignore this comment entirely.

Will the `AllowedSourceIps` always apply only to this one exact VPC ID? For example, if in the future we split `SERVICES_VPC_ID` up for some reason, or if we added some other kind of external service VPC, would we want the rules applied to new things too?

This is probably fine, I'm just wondering if there's a way to structure this that's less likely to be accidentally out of date if we make a change like that. Maybe we could move this check to be a method on `AllowedSourceIps` itself? `AllowedSourceIps::applies_to_vpc(...) -> bool` or something. That might not be any better in this regard.

I have the same concern about looking for `SERVICES_DB_NAME` below, I think. I'm similarly stumped on how to make that better today.

Reply:
This is a good question, which I also debated. The short answer is: I don't know. The more complete answer is that this may definitely apply to more than one VPC in the future. I opted to avoid future-proofing like that because it's both uncertain, and also uncertain if we'll even implement the allowlist through a firewall rule in that case. RFD 474 talks about this a bit, but the idea is that we may want to move the enforcement to the application itself. For example, some DNS names may need to be public even though other services (or other _names_) are not.

That said, I do really like the suggestion of a more explicit interface. It's a bit funny with exactly one allowlist, but perhaps that would let us support more than one in the future. For getting the name, I could imagine that we would use the result of `applies_to_vpc` to determine whether we should then get the _name_ of that VPC and lookup the NICs in it. Something like:

```rust
if allowlist.applies_to_vpc(vpc.id()) {
    let name = vpc.name();
    let interfaces = vpc_interfaces.get(&name).unwrap();
    // Continue with the other work....
}
```

Reply:
I like that proposal. It also seems fine to defer to later. Up to you.

Reply:
Did a bit of this in 381b1d2c0. We don't need the name lookup, so we just have a free function that checks if a VPC needs the allowlist included in its rules.